lib/spack/spack/relocate.py

```diff
@@ -117,6 +177,8 @@ def relocate_binary(path_name, old_dir, new_dir, patchelf_executable):
     orig_rpath = get_existing_rpath(path_name, patchelf_executable)
     new_rpath  = substitute_rpath(orig_rpath, old_dir, new_dir)
     modify_rpath(path_name, orig_rpath, new_rpath, patchelf_executable)
```

Comment:
maybe we can already pull the darwin specific parts out if modify_rpath and put all together in one function for linux and one for mac?